norm_calc:
  ros__parameters:
    # ==================== 区域与网格 ====================
    BOX_LEN: 0.05
    BOX_ROW: 4
    BOX_COLUMN: 6
    # 工作空间
    XMIN: -0.15
    XMAX: 0.15
    YMIN: -0.1
    YMAX: 0.1
    ZMIN: 0.2
    ZMAX: 0.8
    BORDER_WIDTH: 0.03
    
    # ==================== 预处理 ====================
    SEARCH_RADIUS: 0.02        # 法向估计半径
    SEARCH_NUM_TH: 30          # 离群点去除阈值
    
    # ==================== 核心策略参数 ====================
    # 严格模式 (第一次尝试：只打完美的点)
    STRICT_NORM_TH: 0.96       # cos(16度)，要求非常垂直
    STRICT_HOLE_DIST: 0.05     # 离空洞/旧点至少 5cm
    STRICT_CURV_TH: 0.04       # 表面必须非常平整
    
    # 宽松模式 (第二次尝试：允许次优点)
    RELAXED_NORM_TH: 0.86      # cos(30度)，允许斜坡
    RELAXED_HOLE_DIST: 0.035   # 离空洞至少 3.5cm (不能太近，否则崩塌)
    RELAXED_CURV_TH: 0.10      # 允许表面稍微粗糙
    
    # ==================== 评分权重 ====================
    HEIGHT_WEIGHT: 3.0         # 优先打凸起 (Z值大)
    CURV_WEIGHT: 2.0           # 优先打平整
    ANGLE_WEIGHT: 1.0          # 优先打垂直
    CENTER_WEIGHT: 5.0         # 优先打网格中心 (防止边缘效应)
     # ==================== 凸起处理策略参数 ====================
    PROTRUSION_TH: 0.02       # (米) Z轴极差超过2cm，则判定为存在凸起
    TIP_CROP_RATIO: 0.25      # 遇到凸起时，剔除顶部 25% 的高度 (避开滑动的顶点)
    BASE_CROP_RATIO: 0.10     # 剔除底部 10% 的高度 (避开太深的位置)
    